src/main/java/co/rsk/bitcoinj/script/Script.java

```diff
@@ -482,50 +481,46 @@ public Script getScriptSigWithSignature(Script scriptSig, byte[] sigBytes, int i
         return ScriptBuilder.updateScriptWithSignature(scriptSig, sigBytes, index, sigsPrefixCount, sigsSuffixCount);
     }
 
-    private RedeemScriptParser getRedeemScriptParser() {
-        if (redeemScriptParser == null){
-            redeemScriptParser = RedeemScriptParserFactory.get(chunks);
-        }
-        return redeemScriptParser;
-    }
-
     /**
      * Returns the index where a signature by the key should be inserted. Only applicable to
      * a P2SH scriptSig.
      */
     public int getSigInsertionIndex(Sha256Hash hash, BtcECKey signingKey) {
+        List<ScriptChunk> rawChunks = this.getChunks();
         // Iterate over existing signatures, skipping the initial OP_0, the final redeem script
         // and any placeholder OP_0 sigs.
-        List<ScriptChunk> existingChunks = chunks.subList(1, chunks.size() - 1);
-        ScriptChunk redeemScriptChunk = chunks.get(chunks.size() - 1);
-        checkNotNull(redeemScriptChunk.data);
-        Script redeemScript = new Script(redeemScriptChunk.data);
+        List<ScriptChunk> existingChunks = rawChunks.subList(1, rawChunks.size() - 1);
+        RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(this.getChunks());
```

Comment:
`this` is refering to a scriptSig, is that the case?

Why pass all the chunks to the RedeemScriptParser? Shouldn't you be passing just the redeem script chunks?

Reply:
Updated.

Reply:
it doesn't appear updated to me natha

Reply:
🤔